storage/src/node/mod.rs

```diff
@@ -223,12 +223,7 @@ impl Node {
     pub fn as_bytes<T: ExtendableBytes>(&self, prefix: u8, encoded: &mut T) {
         match self {
             Node::Branch(b) => {
-                let child_iter = b
-                    .children
-                    .iter()
-                    .enumerate()
-                    .filter_map(|(offset, child)| child.as_ref().map(|c| (offset, c)));
-                let childcount = child_iter.clone().count();
+                let childcount = b.children.iter().filter(|c| c.is_some()).count();
```

Comment:
Surprised to see the `clone` was slower than creating the iterator twice. Using `as_ref` is actually the preferred way to do this, as this will not advance the iterator and can be used to fetch the count.